wave_matching/include/wave/matching/matcher.hpp

```diff
@@ -0,0 +1,43 @@
+#ifndef __WAVE_MATCHING_MATCHER_HPP__
+#define __WAVE_MATCHING_MATCHER_HPP__
+
+#include <Eigen/Core>
+#include <Eigen/Geometry>
+
+namespace wave {
+namespace matching {
+
+// sensor data type
+template <typename R>
+class Matcher {
+  public:
+    // generic accessors
+    const Eigen::Affine3d& get_result() {return result;};
+    const Eigen::MatrixXd& get_info() {return information;};
+    const float& getRes() {return resolution;};
+    // Constructor with resolution
+    Matcher(float res) : resolution(res) {}
+    Matcher() {resolution = -1;}
+    // intended to put sensor data into matcher class
+    virtual void setRef(const R& ref) = 0;
+    virtual void setTarget(const R& target) = 0;
+    void setup(const R& ref, const R& target) {
+        this->setRef(ref);
+        this->setTarget(target);
+    };
+    // intended to actually perform the matching, and
+    // then populate the private variables.
+    // should return true if successful
+    virtual bool match() {return 0;}
```

Comment:
Typically the method declaration should follow a logical order, with:

1. Constructors
2. Useful methods
3. Setters and Getters

you currently have 
1. Generic acccessors
2. Constructors
3. Generic accessors
4. Useful Methods